enhancements/ingress/set-delete-http-headers.md

```diff
@@ -1001,17 +1000,54 @@ This route can be annotated with `haproxy.router.openshift.io/disable_cookies=tr
   must not exceed the value of spec.tuningOptions.headerBufferMaxRewriteBytes on the
   IngressController.
 
-#### Why does a custom header value defined in RouteSpec override IngressControllerSpec value ?
-  Answer: It's quite common to have a global setting with an escape hatch locally to override the global setting.
-  The purpose of this EP is to define custom headers. So, there are two reasons we need such override:
-  1. A route owner shall have the choice to define what headers he wants as per his application requirements(as they might be varying) 
-  than the ones admin has set cluster-wide.
-  2. The IngressController and Route might be owned by the same person who wants to specify some global configuration
-  and then override that global configuration on a small subset of routes. 
+#### How do Route headers interact with IngressController headers of the same value?
+  Answer:  It depends on if they are response headers or request headers.
+
+In case of HTTP response headers, the actions specified in `spec.httpHeaders.actions` on the IngressController will be executed after
+the actions specified in the Route's `spec.httpHeaders.actions` field.
+
+In the case of HTTP request headers, the actions specified in `spec.httpHeaders.actions[*].request` on the Route will be executed after
+the actions specified in the IngressController's `spec.httpHeaders.actions[*].request` field.
+
+ This is because of the way HAProxy is designed to work. We cannot change the processing order; we're constrained by HAProxy's design: 
+ the frontend receives the request and dispatches it to the backend, and then the frontend gets the 
+ response from the backend.
+
+This means that for requests, [the frontend's rules are applied first, followed by the backend's,
+and conversely](https://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-request), for responses, [the backend's rules are applied first, followed by the frontend's](https://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-response).
+ The `http-response set or delete` stanzas are applied and interpreted in this way by the HAProxy.
+For example: A cluster admin sets a response header with the name `X-Frame-Options` and the value `DENY` using the IngressController API. However, a route owner sets  the `X-Frame-Options` header with the value `SAMEORIGIN`.
+
+In this scenario, `haproxy.config` will look like the following where
```

Comment:
I think you should add a line break before "For example" and remove the line break before "In this scenario".

```suggestion

For example: A cluster admin sets a response header with the name `X-Frame-Options` and the value `DENY` using the IngressController API. However, a route owner sets  the `X-Frame-Options` header with the value `SAMEORIGIN`.  In this scenario, `haproxy.config` will look like the following where
```

Reply:
ok